src/state/messages/index.tsx

```diff
@@ -37,15 +38,18 @@ export function ChatProvider({
       }),
   )
   const service = useSyncExternalStore(convo.subscribe, convo.getSnapshot)
+  const {mutate: markAsRead} = useMarkAsReadMutation()
 
   useFocusEffect(
     React.useCallback(() => {
       convo.resume()
+      markAsRead({convoId})
 
       return () => {
         convo.background()
+        markAsRead({convoId})
```

Comment:
These handlers in this file work like this:
- when a chat is focused, it's marked as read because it's active
- upon blurring a chat, it's marked as read because we want to mark it read up to the last point it was active
- if the app is foregrounded and the screen is focused, it's marked as read because it's active
- if the app is backgrounded (or tab is unfocused) and the screen was focused, it's marked as read because we want to mark it read up to the last point it was active